ipywidgets/__init__.py

```diff
@@ -1,7 +1,11 @@
 from .widgets import *
 
 # Register a comm target for Javascript initialized widgets..
-from IPython import get_ipython
-ip = get_ipython()
-if ip is not None and hasattr(ip, 'kernel') and hasattr(ip.kernel, 'comm_manager'):
-    ip.kernel.comm_manager.register_target('ipython.widget', Widget.handle_comm_opened)
+def handle_kernel(kernel):
+    kernel.comm_manager.register_target('ipython.widget', Widget.handle_comm_opened)
```

Comment:
I think relying on the shell calling handle_kernel should be a fallback. We should probably still have this logic at import, since we want it to be the way of the future, anyway.

So leave `handle_kernel`, but call also call it on import.